inline/src/inlines/types/plain.rs

```diff
@@ -0,0 +1,59 @@
+//! Structs and implementations of Inlines that can contain only String as content.
+
+use unimarkup_commons::scanner::span::Span;
+
+macro_rules! impl_plain_inline {
+    (
+        $(
+            $(#[$($attrss:tt)*])*
+            $inline:ident
+        ),*
+    ) => {
+    $(
+        $(#[$($attrss)*])*
+        #[derive(Debug, Clone, PartialEq, Eq)]
+        pub struct $inline {
+            pub(crate) content: String,
+            pub(crate) span: Span,
+        }
+
+        impl $inline {
+            /// Returns the length of this inline.
+            pub fn content_len(&self) -> usize {
+                self.content.len()
+            }
+
+            /// Returns immutable reference to inner content.
+            pub fn inner(&self) -> &str {
+                &self.content
+            }
+
+            /// Returns mutable reference to inner content.
+            pub fn inner_mut(&mut self) -> &mut str {
+                &mut self.content
+            }
+
+            #[doc = concat!(r"Appends another ", stringify!($inline), " inline to this one.")]
+            pub fn append(&mut self, other: Self) {
+                self.content.push_str(&other.content);
+                self.span.end = other.span.end;
+            }
+        }
+    )*
+    };
+}
+
+impl_plain_inline!(
+    /// A plain text inline.
+    Plain,
+    /// A verbatim inline with input preserved.
+    Verbatim,
+    /// Content inside of pair of parentheses.
+    Parentheses,
+    /// Newline literal.
+    Newline,
+    /// Any whitespace literal except newline.
+    Whitespace,
+    /// Inline explicitly marking end of line.
```

Comment:
That would make lexing a little simpler IMO. Right now, consecutive plain symbols are consumed into single `Token` with `TokenKind::Plain`. This consumes whitespace as well. But this was done at lexing stage previously because of `String` allocations, larger plain tokens would reduce number of allocations at parsing stage. But there are no allocations at lexing stage now, so we can do this differently. 

My suggestion would be, if lexer does not recognize symbol as a keyword, substitution or an escape sequence, we can simply wrap the symbol into plain token and pass it to parser. Parser can then decide whether whitespace should be preserved or merged. 🤔

Reply:
I still didn't name them as I should've. Here's how will that look like now: 

```
Newline -> just regular newline
Whitespace -> any regular whitespace
EscapedNewline -> Newline that has been explicitly escaped (e.g. `\\n`)
EscapedWhitespace -> Whitespace that has been explicitly escaped (e.g. `\\t`)
```

This will now be same for `TokenKind` and `Inline`.